state-transition/core/validation_deposits.go

```diff
@@ -122,7 +122,7 @@ func ValidateNonGenesisDeposits(
 	}
 
 	// Finally check that the historical deposits root matches locally what's on the beacon block.
-	if !localDeposits.HashTreeRoot().Equals(blkDepositRoot) {
+	if !depRoot.Equals(blkDepositRoot) {
```

Comment:
nit naming for clarity:
```suggestion
	if !localDepositRoot.Equals(blkDepositRoot) {
```